calibr8/tests.py

```diff
@@ -460,6 +460,11 @@ def test_symbolic_loglikelihood(self):
         x_true = numpy.array([1, 2, 3, 4, 5])
         y_obs = cmodel.predict_dependent(x_true)[0]
 
+        # Make sure that one can't logp graphs that depend on a random variable
+        with pytest.raises(ValueError, match="Random variables detected"):
+            xRV = pm.Normal.dist()
```

Comment:
Could you explain to me how this leads to an error but using `self.pymc_dist.dist` is not? I think this now went quite deeply into PyMC specifics and I might not be up to date concerning logP

Reply:
The difference here is a matter of creating PyMC model variables, or creating raw PyTensor random variables via the `.dist` convenience API provided by a PyMC distribution.

A model variable is associated with extra stuff like automatic transforms, whereas a pure random variable is not. The `xRV` here is what you'd also get from calling `pt.random.normal()`:

```python
>>> import pytensor.tensor as pt 
>>> pt.random.normal()
normal_rv{0, (0, 0), floatX, False}.out
>>> import pymc as pm
>>> pm.Normal.dist()
normal_rv{0, (0, 0), floatX, False}.out
```

If you were to create a logp as done in this test, you'd get something where `xRV` just takes on random value that doesn't even change if you re-evaluate the graph.

What we actually want here is a graph where we have to provide the value for `x` as the input. This is what we do a few lines below, by creating a `x_hat = at.vector()` and providing a value for it in `L.eval({x_hat: x_test})`.

Does that answer your question?

Reply:
Yes